ref/haskell/src/Codec/Binary/Bech32.hs

```diff
@@ -156,4 +177,14 @@ segwitDecode hrp addr = do
 segwitEncode :: HRP -> Word8 -> Data -> Maybe BS.ByteString
 segwitEncode hrp witver witprog = do
     guard $ segwitCheck witver witprog
-    bech32Encode hrp $ UnsafeWord5 witver : toBase32 witprog
+    rightToMaybe $ bech32Encode hrp $ UnsafeWord5 witver : toBase32 witprog
```

Comment:
I think `foo . bar $ x` is preferred to `foo $ bar $ x` generally speaking.  But this isn't show-stopping.